pkg/trait/mount.go

```diff
@@ -0,0 +1,199 @@
+		if !strings.HasPrefix(c, "configmap:") && !strings.HasPrefix(c, "secret:") {
+			return false, fmt.Errorf("unsupported config %s, must be a configmap or secret resource", c)
+		}
+	}
+	for _, r := range t.Resources {
+		if !strings.HasPrefix(r, "configmap:") && !strings.HasPrefix(r, "secret:") {
+			return false, fmt.Errorf("unsupported resource %s, must be a configmap or secret resource", r)
+		}
+	}
+
+	return true, nil
+}
+
+func (t *mountTrait) Apply(e *Environment) error {
+	if e.IntegrationInPhase(v1.IntegrationPhaseInitialization) {
+		return nil
+	}
+
+	container := e.GetIntegrationContainer()
+	if container == nil {
+		return fmt.Errorf("unable to find integration container: %s", e.Integration.Name)
+	}
+
+	var volumes *[]corev1.Volume
+	visited := false
+
+	// Deployment
+	if err := e.Resources.VisitDeploymentE(func(deployment *appsv1.Deployment) error {
+		volumes = &deployment.Spec.Template.Spec.Volumes
+		visited = true
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	// Knative Service
+	if err := e.Resources.VisitKnativeServiceE(func(service *serving.Service) error {
+		volumes = &service.Spec.ConfigurationSpec.Template.Spec.Volumes
+		visited = true
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	// CronJob
+	if err := e.Resources.VisitCronJobE(func(cron *v1beta1.CronJob) error {
+		volumes = &cron.Spec.JobTemplate.Spec.Template.Spec.Volumes
+		visited = true
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if visited {
+		// Volumes declared in the Integration resources
+		e.configureVolumesAndMounts(volumes, &container.VolumeMounts)
+		// Volumes declared in the trait config/resource options
+		err := t.configureVolumesAndMounts(e, volumes, &container.VolumeMounts)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *mountTrait) configureVolumesAndMounts(e *Environment, vols *[]corev1.Volume, mnts *[]corev1.VolumeMount) error {
+	for _, c := range t.Configs {
+		if conf, parseErr := utilResource.ParseConfig(c); parseErr == nil {
+			t.attachResource(e, conf)
+			t.mountResource(vols, mnts, conf)
+		} else {
+			return parseErr
+		}
+	}
+	for _, r := range t.Resources {
+		if res, parseErr := utilResource.ParseResource(r); parseErr == nil {
+			t.attachResource(e, res)
+			t.mountResource(vols, mnts, res)
+		} else {
+			return parseErr
+		}
+	}
+	for _, v := range t.Volumes {
+		if vol, parseErr := utilResource.ParseVolume(v); parseErr == nil {
+			t.mountResource(vols, mnts, vol)
+		} else {
+			return parseErr
+		}
+	}
+
+	return nil
+}
+
+// attachResource is in charge to filter the autogenerated configmap and attach to the Integration resources.
+// The owner trait will be in charge to bind it accordingly.
+func (t *mountTrait) attachResource(e *Environment, conf *utilResource.Config) {
+	if conf.StorageType() == utilResource.StorageTypeConfigmap {
+		// verify if it was autogenerated
+		cm := kubernetes.LookupConfigmap(e.Ctx, e.Client, e.Integration.Namespace, conf.Name())
```

Comment:
I'm realising those new calls to `kubernetes.LookupConfigmap` will start watching ConfigMap resources, possibly on the entire cluster for the global setup. This should ideally be avoided. One option is to use the _non-caching_ client, or work with unstructured ConfigMap, as caching is disabled for unstructured objects by default.